### ISSUE TITLE
POST requests with params to the Graph API, are they possible?

### ISSUE THREAD

**Issue by generagames** [myflashlab/facebook-ANE#44]
Hello,

We recently got your Facebook ANE, and we were wondering about if it supports **POST requests with params** to the Graph API, as we want to use Custom Open Graph stories to publish data on behalf of the user.

We have received feedback for your contact link by email telling us it's indeed possible, and as I see in V4.11.0 (the one we own), there's a request() method in the Graph class to work with the Facebook Graph API, with a $method param that you can use to specify the nature of your request (POST in our case)... but **I see no way to add POST params apart from adding them to the $msg URL** as you would do in a GET request.

If it's possible, I will really appreciate if you can write me a quick example of a **/user/feed** publish operation filling the **message** param using your extension ( https://developers.facebook.com/docs/graph-api/reference/v2.6/user/feed#publish )

Best Regards,

Carlos.

**Comment by myflashlab** [comment]
I admit that it's a bit confusing. We are planning to change this behavior so it won't be misleading. but I also confirm that you can send POST by the ```request``` method.

you see, when you are changing the second parameter of the ```request``` method from GET to POST, your variables will be sent in POST method even though you are formatting them like ```...?param1=value1&param2=value2```

the ANE will automatically turn the parameters into the suitable format based on the request method. if it's POST, they will be sent in POST. So, you'll be fine.

*note: I am marking this ticket as an enhancement so the future version of the ANE will have a clear approach toward sending POST messages.*

**Comment by generagames** [comment]
Ah I see! Sorry didn't think about that.

The problem is that it may be difficult to encode complex requests in the Query String (and URL encoding is a completely headache). And also [posting photos](https://developers.facebook.com/docs/graph-api/reference/user/photos/) may not be possible with this, no?

Thanks for your response! Best regards.

**Comment by myflashlab** [comment]
Yes, totally agree with you. it's a bit confusing to send requests like that. we will be changing this by mid June and will release a new version.

about posting photos, I think you can use something like this:
```html
POST /v2.6/me/photos HTTP/1.1
Host: graph.facebook.com

url=%7Bimage-url%7D
```

anyway, next version will have this feature boosted :) we'll make it work like how the [REST API works in our Games Services ANE](http://myflashlab.github.io/asdoc/com/myflashlab/air/extensions/gameServices/google/restApi/ApiRest.html)

**Comment by generagames** [comment]
Thanks! I'm looking forward to getting the next version :).

Hehe, that's it, you could specify an image that is image that is already hosted in some server and specify its URL, but I think you won't be able to upload a ByteArray/Bitmap... right? Here is the uploading photos section in Facebook:

_There are two separate ways of publishing photos to Facebook:_

_1: **Attach the photo as multipart/form-data**. The name of the object doesn't matter, but historically people have used source as the parameter name for the photo. How this works depends on the SDK you happen to be using to do the post._

_2: Use a photo **that is already on the internet** by publishing using the url parameter:_

```
POST /v2.6/me/photos HTTP/1.1
Host: graph.facebook.com

url=%7Bimage-url%7D
```

If we could find this features in the next version it would be great :).

Regards.

**Comment by myflashlab** [comment]
This request has been addressed with the upcoming release of V4.17.0 in a couple of hours. Please check the changelog in a few hours to see how you can use the new method. here's a preview:

```actionscript
FB.graph.addEventListener(FBEvent.GRAPH_RESPONSE, onGraphResponse);
FB.graph.addEventListener(FBEvent.GRAPH_RESPONSE_ERROR, onGraphError);
FB.graph.call("https://graph.facebook.com/v2.8/me", URLRequestMethod.GET, new URLVariables("fields=name,email,picture&metadata=0"));
```
